### PR TITLE
Feature bubble sort project

### PR DESCRIPTION
**Pull request by atoledo-ec**
## Feature Bubble Sort Project

This project was build with RUBY.

This project contains two methods:
- First Method: sorts the number inside an array
- Second Method: sorts words inside and array

### CODE REVIEW

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

#### Hey team,  your project is looking really good, however, you're required to do these changes, listed below:
  
## [REQUIRED CHANGES]
- [ ] Your linters is throwing errors, please fix it;
- [ ] Your `bubble_sort_by` method is also throwing errors when users try to pass an empty bock through it. Tip:  you can fix it by using yield
image
- [ ] Improve your readme descriptiveness, like at the setup section. Describe what the user should do to see what you did in this project. Assume that the user doesn't have any knowledge in ruby or programming languages, it will help you when you be required to write some documentation, for example.

## Please make the required changes and resubmit for a new code review once you are ready💪

## As always, I wish you all the very best and see you both at the top🚀


👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In real-world, you'll have a lot of changes requests for your real clients, it's good to get used to it.</sub>

File: bubble_sort.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    no_swapped = false
+    (1...n).each do |i|
+      # compare the length of the strings into the array
+      if arr[i - 1].length > arr[i].length
+        arr[i - 1], arr[i] = arr[i], arr[i - 1]
+        no_swapped = true
+      end
+    end
+  end
+  yield p arr
```
Comment (acushlakoncept) [inline bubble_sort.rb]: An error occurred while loading ./bubble_sort_spec.rb.
`Failure/Error: yield p arr`

Kindly check this line of code to fix the issue.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

Hello @julian3493 and @alejandrotoledoweb, Great job! 👏  so far.
Let's make a few changes, and we will be done.

- [ ] You are yet to fix the error requested by the previous TSE.  You can use `block_given?` to check if block is given or not.
Screen Shot 2020-10-14 at 11 38 12 PM

- [ ] Kindly check the comment below as well.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for making the suggested changes
However there is still some bugs

1. Look at this, please
```rb
Traceback (most recent call last):
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        3: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        2: from (irb):1
        1: from (irb):1:in `load'
SyntaxError (bubble_sort.rb:28: syntax error, unexpected '=')
        var = arr[i] < = > arr[i - 1]
                       ^
2.6.5 :002 > 
```
Kindly fix it by using `<=>`, remove space

2- Also, raises an error if block is not given to the `bubble_sort_by` method.

3- Finally, kindly fix the linter's error

Kindly make the suggested changes and once done submit a new code review request
Thanks

Mael FOSSO

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUIRED ♻️ 

Hello team! 
Good job with the implemented changes! 👏 
But we still have a problem with the bubble_sort_by method when no block is given, in that situation the method should raise an error.
Kindly fix the issue and resubmit your project for a new code review.

Good luck 😺

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__Although we still have some changes to make, you have done an excellent job on this project. Kindly check my comments below and fix the issues as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] Your algorithms (methods) should `return` and `not print` the sorted array. Kindly remove the `p` and just return the sorted array.



- [x] Your `bubble_sort_by` does not pass all the test cases.
- As mentioned by the previous reviewers, your **`bubble_sort_by raises an error if block is not given`.** This means that when your `bubble_sort_by` is tested with no block, it is expected to return an error `LocalJumpError`, but it does not.

If I test your `bubble_sort_by` with the code snippet below:

`bubble_sort_by(%w[hello hi hey])`

This should return an error because the `bubble_sort_by` should only work when a block is given. 

But in your case, it does not return an error. The fact that it doesn't return an error is something you should look into.

**[ COMMENTS ]**
- I can see that you have some inline comments on your project. To me, I think these comments are unnecessary. This is because the code itself is self-explanatory. If you think the same way too, you can remove those comments.

**Please make the required changes and submit another review request when you are done.**

* * *

**_You can reach out to me on twitter (@ola_lere), Github(@teekaytech), or slack (@taofeekolalere )_**
__Happy Coding :smiley:__

**Review by teekaytech — APPROVED** [review]
# Status: Approved :green_circle: 

### Congratulations team :tada:  this project has passed all the mandatory requirements. It is therefore approved :heavy_check_mark:

Do not forget to fill the **project completion form.**
Also, remember to **merge this branch to master.**
_Good luck with your next task._

**I'm @taofeekolalere on slack.
Happy Coding :man_technologist:**

**Pull request merged**